### DISCUSSION TITLE
Roadmap for next year (year 2)?

### DISCUSSION THREAD

**Discussion by leossmith** [airsdk/Adobe-Runtime-Support#597] [Q&A]
We had a nice good 1st year, with a lot of bugs fixed and many features released. 

What should we expect for years 2? 

Can @ajwfrost post a roadmap or what are the plans for next year or at least q1 and q2?

**Comment by MuradHuniti** [comment]
A very good question. I'm really afraid of AIR and Actionscript future, but i hope AIR will flourish under Andrew and HARMAN rule :dancers: 

They are really doing a very nice job, and i hope they gonna answer

**Comment by ajwfrost** [comment]
Sorry yes I meant to answer earlier! I started a response at the weekend but then didn't have the information to hand..

We've basically taken on board a number of the requests and suggestions that people have provided to us, so are trying to move this in the direction where it will be of most benefit to the largest number of developers. It still actually feels like we are being too reactive currently - i.e. we're primarily focused on fixing bugs and reacting to changes in the OS requirements from Google and Apple - but the team is keen to do some 'new' development too!

We do plan on this being an offering for the long term - so then in terms of the next steps, we've got:
- multimedia support (including starting to enhance this to support newer formats/protocols)
- localisation updates (additional language choices for application descriptors etc)
- toolchain improvements (ADT commands, hopefully Android App Bundle)
- stability improvements
- device display information (new classes to find out screen info, cut-out details, and control status bars etc)
- OS compatibility (next Android targetSDK, iOS 14, MacOS 11 and ARM-based Mac)

and we're looking at Linux and web targets which is a slightly longer term play.
And we'd definitely like to reduce the number of bugs that are being reported!

There's also a small team working on improvements to the website, currently doing some stuff on the back end and our admin components but we're also looking to roll out things like the ability to download some of the earlier releases and then will look at hosting the AS3 documentation and a lot of the Adobe dev-net articles, and getting some discussion board going (although this works pretty well I think!)

Always open to ideas of what would be useful for developers both in terms of AIR features, and in terms of improving the workflow..

thanks!

**Comment by Michoko92** [comment]
Support for Adaptive Icons on Android would be nice, please. :)

**Comment by Elintondm** [comment]
Would be great if Adobe Air could be deployed as Html5 or any Browser
compatible way. We are very worried about the end of SWF support of
browsers.

Adobe Air is still a great platform to build 2d games, but will be hard to
continue without browser support.

You are doing a good job till now and we hope you can achieve this!

Em sáb., 18 de jul. de 2020 às 10:30, Michoko <notifications@github.com>
escreveu:

**Comment by marchbold** [comment]
[Custom resources for Android](https://github.com/Gamua/Adobe-Runtime-Support/issues/163) would be a massive improvement, removing the need for devs to create a custom ANE just to include some configuration (firebase values) or image assets (notification icons etc).

**Comment by Mintonist** [comment]
Web assembly target can be a revolution!

https://github.com/Gamua/Adobe-Runtime-Support/issues/163 +1

**Comment by MuradHuniti** [comment]
Custom resources for Android is the first thing HARMAN should do, some ANE's are 10x expensive than AIR it self. What a logic !!!!
@marchbold

**Comment by tidus2020** [comment]
My wish list:
1,easy to build ANE or call the native api without ANE.
2,target to html5
3,some improvement for 3d
~~~o((≧▽≦o)

**Comment by hoangdung-qbt** [comment]
My wishlist: 
1. add adt cl that can create signed apps for Mac Catalina and Mac App Store.
2. Fix "lost context" bug (or reduce it) when reactivating apps on Android.

**Comment by al-sabr** [comment]
Can you guys just open source the AIR so that we can all contribute to it?

**Comment by ajwfrost** [comment]
@al-sabr  we can't I'm afraid: it remains software that we license from Adobe, with very strict confidentiality requirements around the source code. Potentially if there are features that could be written in a 'stand alone' fashion which we could hook into dynamically, then it would become something that more folk could contribute to. So e.g. if anyone fancies writing a new implementation of the Stage3D APIs using Vulkan, we may be able to integrate that into AIR..?! :-)

@hoangdung-qbt: we are looking at updating the ADT and sorting out signing/certificates so that it's possible to create signed .app and .dmg packages and getting them all ready for notarisation.. so that should be viable. The "lost context" we had a quick look at a while ago and is more down to the notifications we get from the OS.

@tidus2020 I had been wondering if it's viable to have a native code mechanism more similar to JNI... or perhaps providing a library loading/marshalling mechanism like in C# .. would be interested in your thoughts of how you'd like to see this work!

Custom resources for Android - which hopefully enables adaptive icons - is partially done in 33.1.1.259 and should be updated slightly for the next release.

We're also looking at the 'shared' AIR runtime for desktops and some updates to ensure application deployment is a little easier on desktop platforms.

**Comment by leossmith** [comment]
@ajwfrost I was wondering if it is possible to have something like Hot Reload in AIR. This is such a time saver during development.

Basically Hot Reload works as:
1. Start Debugger (running app on adl, emulator or device)
2. Do any changes in code.
3. Save
4. SWF updates with the latest changes and the app remains to the screen / view / wherever you are.

This is one of the biggest benefits during development in Flutter which made me think on how much I am spending editing apps and waiting for them to rerun so I can check even the tiniest change.

I was just trying to figure out if it is even possible to do in AIR.

**Comment by al-sabr** [comment]
@leossmith This is already possible with the OSGi Framework ADEP Gravity Client Components framework.

**Comment by leossmith** [comment]
But this isn't really a solution as it only applies if you use this specific framework. I am talking a complete AIR Hot Reload
